mappings/net/minecraft/util/math/Box.mapping

```diff
@@ -32,6 +38,14 @@ CLASS net/minecraft/class_238 net/minecraft/util/math/Box
 	METHOD method_1005 getCenter ()Lnet/minecraft/class_243;
 	METHOD method_1006 contains (Lnet/minecraft/class_243;)Z
 		ARG 1 vec
+	METHOD method_1007 traceCollissionSide (Lnet/minecraft/class_238;Lnet/minecraft/class_243;[DLnet/minecraft/class_2350;DDD)Lnet/minecraft/class_2350;
```

Comment:
`traceCollisionSide`
Also, what does this method do exactly ? I can't really tell just by looking at the name.

Reply:
Best I can say is it it's used for rayTracing to find either the side of the box the ray enters, or that the ray exits.

The `rayTrace` method calls this once for each side, every time passing in a shared double[] array and the direction produced by the last call of the method.

    @Nullable
    public static BlockHitResult rayTrace(Iterable<Box> boxes, Vec3d from, Vec3d to, BlockPos blockPos) {
        double[] doubles_1 = { 1.0 };
        Direction direction6 = null;
        double double7 = to.x - from.x;
        double double9 = to.y - from.y;
        double double11 = to.z - from.z;
        for (final Box class_238_1 : boxes) {
            direction6 = traceCollissionSide(class_238_1.offset(blockPos), from, doubles_1, direction6, double7, double9, double11);
        }
        if (direction6 == null) {
            return null;
        }
        double double13 = doubles_1[0];
        return new BlockHitResult(from.add(double13 * double7, double13 * double9, double13 * double11), direction6, blockPos, false);
    }
    
    @Nullable
    private static Direction traceCollissionSide(Box box, Vec3d approachVector, double[] traceDistanceResult, @Nullable Direction approachDirection, double xDelta, double yDelta, double zDelta) {
        if (xDelta > 1.0E-7) {
            approachDirection = traceCollissionSide(traceDistanceResult, approachDirection, xDelta, yDelta, zDelta, box.minX, box.minY, box.maxY, box.minZ, box.maxZ, Direction.WEST, approachVector.x, approachVector.y, approachVector.z);
        }
        else if (xDelta < -1.0E-7) {
            approachDirection = traceCollissionSide(traceDistanceResult, approachDirection, xDelta, yDelta, zDelta, box.maxX, box.minY, box.maxY, box.minZ, box.maxZ, Direction.EAST, approachVector.x, approachVector.y, approachVector.z);
        }
        if (yDelta > 1.0E-7) {
            approachDirection = traceCollissionSide(traceDistanceResult, approachDirection, yDelta, zDelta, xDelta, box.minY, box.minZ, box.maxZ, box.minX, box.maxX, Direction.DOWN, approachVector.y, approachVector.z, approachVector.x);
        }
        else if (yDelta < -1.0E-7) {
            approachDirection = traceCollissionSide(traceDistanceResult, approachDirection, yDelta, zDelta, xDelta, box.maxY, box.minZ, box.maxZ, box.minX, box.maxX, Direction.UP, approachVector.y, approachVector.z, approachVector.x);
        }
        if (zDelta > 1.0E-7) {
            approachDirection = traceCollissionSide(traceDistanceResult, approachDirection, zDelta, xDelta, yDelta, box.minZ, box.minX, box.maxX, box.minY, box.maxY, Direction.NORTH, approachVector.z, approachVector.x, approachVector.y);
        }
        else if (zDelta < -1.0E-7) {
            approachDirection = traceCollissionSide(traceDistanceResult, approachDirection, zDelta, xDelta, yDelta, box.maxZ, box.minX, box.maxX, box.minY, box.maxY, Direction.SOUTH, approachVector.z, approachVector.x, approachVector.y);
        }
        return approachDirection;
    }
    
    @Nullable
    private static Direction traceCollissionSide(double[] traceDistanceResult, @Nullable Direction approachDirection, double xDelta, double yDelta, double zDelta, double begin, double minX, double maxX, double minZ, double maxZ, Direction resultDirection, double startX, double startY, double startZ) {
        double double26 = (begin - startX) / xDelta;
        double double28 = startY + double26 * yDelta;
        double double30 = startZ + double26 * zDelta;
        if (0.0 < double26 && double26 < traceDistanceResult[0] && minX - 1.0E-7 < double28 && double28 < maxX + 1.0E-7 && minZ - 1.0E-7 < double30 && double30 < maxZ + 1.0E-7) {
            traceDistanceResult[0] = double26;
            return resultDirection;
        }
        return approachDirection;
    }